### PR TITLE
Add InvalidSignatureException to supported error types

### PR DESCRIPTION
**Pull request by bobwanglyft**
Commit Message: Add `InvalidSignatureException` to supported error types and allow error type being captured by stats.
Risk Level: Low

Signed-off-by: Bob Wang bobwang@lyft.com

### CODE REVIEW

**Comment by dio** [comment]
Thanks, @bobwanglyft. Could you [fix DCO](https://github.com/envoyproxy/envoy/blob/8b6abe6293cced29a0dd63d2aa6dea15702a3512/CONTRIBUTING.md#fixing-dco)?

File: source/extensions/filters/http/dynamo/dynamo_request_parser.cc [hunk 1]
```diff
@@ -39,6 +39,7 @@ const std::vector<std::string> RequestParser::SUPPORTED_ERROR_TYPES{
     "ConditionalCheckFailedException",
     "IdempotentParameterMismatchException",
     "IncompleteSignatureException",
+    "InvalidSignatureException",
```
Comment (dio) [inline source/extensions/filters/http/dynamo/dynamo_request_parser.cc:42]: While you at this, could you add a link to dynamo docs providing the list of supported error types?

Comment (bobwanglyft) [reply source/extensions/filters/http/dynamo/dynamo_request_parser.cc:42]: @dio Thanks for reviewing the PR :) 
This error is not listed in the [error handling section](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html#Programming.Errors.MessagesAndCodes) of DynamoDB developer guide, but users do get this error for [various reasons](https://github.com/aws/aws-sdk-go/issues/2598#issuecomment-526398896). What is the recommended way of adding this info into the repo?

Comment (dio) [reply source/extensions/filters/http/dynamo/dynamo_request_parser.cc:42]: Exactly what you have said above will be good as inline comments. Telling that these are the errors listed in the docs and there is this one observed at runtime. Thank you!

File: source/extensions/filters/http/dynamo/dynamo_request_parser.cc [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace Dynamo {
 
 /**
  * Basic json request/response format:
- * http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Appendix.CurrentAPI.html
```
Comment (bobwanglyft) [inline source/extensions/filters/http/dynamo/dynamo_request_parser.cc:19]: old link is no longer valid

File: source/extensions/filters/http/dynamo/dynamo_request_parser.cc (hunk 1 above)
Comment (bobwanglyft) [reply source/extensions/filters/http/dynamo/dynamo_request_parser.cc]: Gotcha. I added the info as inline comments. (plus fixing the obsolete url link)

**Comment by bobwanglyft** [comment]
@dio mind taking a look again? :)

**Review by dio — APPROVED** [review]
Thanks

**Comment by dio** [comment]
cc. @lizan, just in case if you want to take another look.

**Pull request merged**